features/step_definitions/service_steps.rb

```diff
@@ -2,6 +2,10 @@
   provider.services.create! :name => name, :mandatory_app_key => false
 end
 
+Given /^a backend api "([^"]*)" of (provider "[^"]*")$/ do |name, provider|
+  provider.backend_apis.create!(:name => name, :private_endpoint => 'https://foo')
+end
```

Comment:
I propose this to avoid being so redundant with the url of the provider that does not even matter:
```
Given /^a backend api "([^"]*)")$/ do |name|
  @provider.backend_apis.create!(:name => name, :private_endpoint => 'https://foo')
end

Given /^(?:a )?default service of has name "([^"]*)"$/ do |name|
  @provider.first_service!.update_attribute(:name, name)
end
```

Reply:
I can't be sure that would be backwards compatible. For this particular cucumber and for the step I added it would be, however keep in mind this is used by a massive number of old features.. it's kinda risky

Reply:
not risky because it is adding new stuff without removing anything :smile:

Reply:
Right 🤦‍♂